### ISSUE TITLE
Improve the docs of commit message guideline 

### ISSUE THREAD

**Issue by stevemao** [ajoslin/conventional-changelog#53]
#52

In the up coming new release we can fully customise the commit message format so a strict commit message guideline is not necessary.

**Comment by stevemao** [comment]
Or we can come up with some non-strict commit message guideline

**Comment by voxpelli** [comment]
I think the strict conventions are one of the pro's with this tool – an easy guideline/standard for your team on how to format commit messages. Even if other alternatives may become supported by this tool I would like to keep the option of enforcing the existing `CONVENTIONS.md` and to keep the file as a reference for that enforcing.

**Comment by stevemao** [comment]
You can still config to make it strict. It just gives people more options and more projects can adapt this.

**Comment by voxpelli** [comment]
@stevemao Like JSCS it would be nice to still have the old style like a "preset" or such: http://jscs.info/overview.html#presets

**Comment by stevemao** [comment]
:+1: I was thinking about that too

**Comment by bachp** [comment]
I also think that having a fixed convention is an advantage of conventional changelog.
And I belive the current format is flexible enough to be working for most people. Choice for the sake of choice is not always good.

**Comment by stevemao** [comment]
OK let me rephrase. I'm not saying there is not going to be strict rules. There are but you can define them. Just like jscs or jshint, you can define the rules. They never restrict you from having your own rules. Same to this project, you don't have to use Angular commit message. If you have your own conventions you can still use this one to generate logs for you based on your commit convention.

**Comment by bachp** [comment]
I understand what you mean.
But I think there should add least be a CONVENTIONS.md document describing a best practice standard.
If configuration options are a good or bad thing is another question. But one example I like to mention is go the programming language, where a simple command [go fmt](https://blog.golang.org/go-fmt-your-code) made the whole discussion about coding style a non issue as there is only one style enforced by an automatic tool.
If there is one convention for conventional-changelog people would not need to do things differently for each project they do.

**Comment by stevemao** [comment]
Sorry, CONVENTIONS.md doesn't have to be removed. It probably needs to be renamed and reworded. Point is we need new docs to guide users.

I understand that you like to have consistency across all projects but in real world each project has their own requirements. If you look at other issues in the repo you'll know why I need to make this more flexible.

**Comment by bachp** [comment]
I'm not oposed to make you tool more flexible, it is great and I think to make it useful for more people is a good thing.

So let me rephrase my proposal:
CONVENTIONS.md (or whatever it's name will be) should contain sane defaults that could be adopted by most projects without any customization. So if I want to use it in my project I can just put a link to this document and that should be enough to tell people how to do commits for their project.

People who need custom formats can still do it but they need to write more documentation themselves ;)

btw: What is the issue with the current CONVENTIONS.md? I would like to help make things better.

**Comment by stevemao** [comment]
Thanks @bachp , Let's wait until #51 is solved. All assumptions here are based on the upcoming release. I'm currently rewriting it. Won't be soon I guess :(.

**Comment by stevemao** [comment]
I have pushed the rewrite to the master branch. Would be great if you could have look. Thanks.

**Comment by huerlisi** [comment]
Could you re-instantiate the `CONVENTIONS.md` file please. Probably just pointing to the new `conventions` directory.

It was referenced in places like https://github.com/btford/grunt-conventional-changelog/blob/master/README.md that now point to no-where.

**Comment by stevemao** [comment]
@huerlisi fixed. Note the grunt wrapper is still using the v0.0.x version of this module and it is too early to update it now.

**Comment by ajoslin** [comment]
It's extremely difficult to get started now. There needs to be an established, opinionated set of defaults that are easy to switch from.

I need to be able to find this repo and get started as quickly as possible, with as little reading and weighing of choices as possible.

A set of conventions needs to be established that work for most people, then the configuration kicks in for everyone else.

"Convention over configuration", they call it. Let's stick with the angular conventions being advertised as the "suggested convention".

If I arrive at this repository and view the README, I should, within a few lines of reading, be able to gather the knowledge necessary to get starte.d

**Comment by ajoslin** [comment]
In fact, I think creating a separate repo called `standard-changelog`, which is a highly opinionated version of `conventional-changelog`, would be a great idea.

Something you can download and start using in 2 minutes or less.

**Comment by stevemao** [comment]
Hi @ajoslin , if you want to use a standard angular conventions you could just add `-p angular` flag to the cli. People can just add that command to their package.json.

The hard part of the config I think is not the preset itself (in fact, it's pretty easy to add a preset). The hard part is if you want to fully customise it. For example, If you want to config other things such as the template (some people want support for bitbucket, jira etc), do wired things with `git log` and a more tolerant commit format. I mean even if people are using angular convention, if they are not on github, they can't generate what they want without these options. Exposing these options in the modules are not too hard (well, maybe a bit of trouble but it's not too bad, really) plus they are valid proposals. 

I know you might say people should use it without being able to config too much and as it should be opinionated but there were a lot of issues asking for these options. My solution to this was:
For most users, if you just do `conventional-changelog` it will generate a log without any particular format (for people who are not using any convention) and if you do `conventional-changelog -p angular` it will generate based on angular convention. Side note: At the moment the presets template is probably best for github but it can grab a few config from [hosts](https://github.com/ajoslin/conventional-changelog/tree/master/hosts) (not too many config for hosts at the moment and I leave this to people who wants more #80). 

So if you just use angular commit convention (and most likely host it on github) you could just do `conventional-changelog -p angular`. I don't know if the extra typing `-p angular` is worth another module?

Also, welcome back to this module Andrew :)

**Comment by ajoslin** [comment]
I think the first thing to address, then, is documentation.

This module is highly customizable, but very difficult to use.

If you approach the readme as a first time user wanting to just generate a changelog, it's very difficult.

The whole "Why" section is questionable. As a user, I honestly don't care about the performance of my changelog generation.

I'll open a PR with my suggestions for the README.

**Comment by ajoslin** [comment]
Opened #82
